### PR TITLE
ss-292: Add api to get a user profile information

### PR DESCRIPTION
**Pull request by Holovashcenko**
image
image

### CODE REVIEW

File: backend/src/bundles/users/user.repository.ts [hunk 1]
```diff
@@ -21,6 +22,21 @@ class UserRepository implements Omit<IRepository, 'update' | 'delete'> {
         return UserEntity.initialize(user);
     }
 
+    public async findUserInfo(
```
Comment (VItaliy-Kalenichenko) [inline backend/src/bundles/users/user.repository.ts:25]:
```suggestion
    public async getCurrentUserDetails(
```

File: shared/src/bundles/users/enums/users-api-path.enum.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 enum UsersApiPath {
     ROOT = '/',
+    GET_ONE_USER = '/user',
```
Comment (VItaliy-Kalenichenko) [inline shared/src/bundles/users/enums/users-api-path.enum.ts:3]:
```suggestion
    GET_USER = '/user',
```

Comment (Holovashcenko) [reply shared/src/bundles/users/enums/users-api-path.enum.ts]: Fixed in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/300/commits/a53e5222a866e9a7e23099d87f575f07e895f32d](url)

File: backend/src/bundles/users/user.repository.ts (hunk 1 above)
Comment (Holovashcenko) [reply backend/src/bundles/users/user.repository.ts]: Fixed in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/300/commits/2bb73852884cc8f060d7f91bffc5474f3f439b69](url)

**Review by anton-otroshchenko — DISMISSED** [review]

**Review by Evcat6 — APPROVED** [review]

File: shared/src/bundles/users/types/user-get-user-profile-response-dto.type.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    createdAt: string | null;
+    updatedAt: string | null;
+    userId: string | null;
+    firstName: string | null;
+    lastName: string | null;
+    sex: string | null;
+    dateOfBirth: string | null;
+    language: string | null;
+    currency: string | null;
+};
```
Comment (nikita-remeslov) [inline shared/src/bundles/users/types/user-get-user-profile-response-dto.type.ts:13]:
```suggestion
type UserProfileResponseDto = {
    email: string;
    id: string;
    createdAt?: string;
};
```

Comment (nikita-remeslov) [reply shared/src/bundles/users/types/user-get-user-profile-response-dto.type.ts:13]: make other field not required

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: shared/src/bundles/users/types/user-get-user-profile-response-dto.type.ts (hunk 3 above)
Comment (Holovashcenko) [reply shared/src/bundles/users/types/user-get-user-profile-response-dto.type.ts]: fix in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/300/commits/b86fdf028d4bb36d00ca5b87354b44cbb450e24e](url)

**Comment by VItaliy-Kalenichenko** [comment]
Resolve conflicts

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**